miio/walkingpad.py

```diff
@@ -0,0 +1,275 @@
+
+    @property
+    def calories(self) -> int:
+        """Current calories burnt."""
+        return int(self.data["cal"])
+
+
+class Walkingpad(Device):
+    """Main class representing Xiaomi Walkingpad."""
+
+    @command(
+        default_output=format_output(
+            "",
+            "Power: {result.power}\n"
+            "Mode: {result.mode.name}\n"
+            "Time: {result.walking_time}\n"
+            "Steps: {result.step_count}\n"
+            "Speed: {result.speed}\n"
+            "Start Speed: {result.start_speed}\n"
+            "Sensitivity: {result.sensitivity.name}\n"
+            "Distance: {result.distance}\n"
+            "Calories: {result.calories}",
+        )
+    )
+    def status(self) -> WalkingpadStatus:
+        """Retrieve properties."""
+
+        data = self._get_quick_status()
+
+        # The quick status only retrieves a subset of the properties. The rest of them are retrieved here.
+        properties_additional = ["power", "mode", "start_speed", "sensitivity"]
+        values_additional = self.get_properties(properties_additional, max_properties=1)
+
+        additional_props = dict(zip(properties_additional, values_additional))
+        data.update(additional_props)
+
+        return WalkingpadStatus(data)
+
+    @command(
+        default_output=format_output(
+            "",
+            "Mode: {result.mode.name}\n"
+            "Time: {result.walking_time}\n"
+            "Steps: {result.step_count}\n"
+            "Speed: {result.speed}\n"
+            "Distance: {result.distance}\n"
+            "Calories: {result.calories}",
+        )
+    )
+    def quick_status(self) -> WalkingpadStatus:
+        """Retrieve quick properties. The walkingpad provides the option.
+
+        to retrieve a subset of properties in one call - steps, mode,
+        speed, distance, calories and time. If you can, use this instead
+        of the full status request, as it is way faster.
+        """
+
+        data = self._get_quick_status()
+
+        return WalkingpadStatus(data)
+
+    @command(default_output=format_output("Powering on"))
+    def on(self):
+        """Power on."""
+        return self.send("set_power", ["on"])
+
+    @command(default_output=format_output("Powering off"))
+    def off(self):
+        """Power off."""
+        return self.send("set_power", ["off"])
+
+    @command(default_output=format_output("Locking"))
+    def lock(self):
+        """Lock device."""
+        return self.send("set_lock", [1])
+
+    @command(default_output=format_output("Unlocking"))
+    def unlock(self):
+        """Unlock device."""
+        return self.send("set_lock", [0])
+
+    @command(default_output=format_output("Starting the treadmill"))
+    def start(self):
+        """Start the treadmill."""
+        if not self.status().is_on:
+            raise WalkingpadException(
+                "Can't start the treadmill, it's not turned on - try issuing an 'on' command first"
+            )
```

Comment:
I thought about this initially, but didn't implement because I wasn't sure if it's nice to implement these types of logics into the library. Regardless, changed to include it now - agree it's a nicer way to behave